Test suite errors and warnings (windows)

Env: Win10 Pro (German)
linkml-runtime: main-branch a8b4dd14abab2eca48cd10a99b287077fcc2c55c

```
(.venv) λ pytest
====================================== test session starts =======================================
platform win32 -- Python 3.10.2, pytest-7.0.1, pluggy-1.0.0
rootdir: C:\Users\david\MyProg\linkml-runtime
collected 83 items
...
============== 4 failed, 74 passed, 4 skipped, 1 xfailed, 7 warnings in 28.60s ===============
```

**Errors**
<details>

```
============================================ FAILURES ============================================ 
_____________________________ Issue368TestCase.test_issue_368_enums ______________________________

self = <tests.test_issues.test_issue_368_enums.Issue368TestCase testMethod=test_issue_368_enums>

    def test_issue_368_enums(self):
        """ Test Enum generation """

>       module = compile_python(env.input_path('issue_368.py'))

tests\test_issues\test_issue_368_enums.py:19:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
linkml_runtime\utils\compile_python.py:47: in compile_python
    exec(spec, module.__dict__)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   ModuleNotFoundError: No module named 'tests\\test_issues\\input'

test:23: ModuleNotFoundError
________________________________ Issue6TestCase.test_loc_function ________________________________

self = <tests.test_issues.test_issue_6.Issue6TestCase testMethod=test_loc_function>

    def test_loc_function(self):
        inp = yaml.load(hbreader.hbread(inp_yaml), DupCheckYamlLoader)
        self.assertEqual('File "<unicode string>", line 3, col 8: ', TypedNode.yaml_loc(inp['foo']['x']))
        self.assertEqual('File "<unicode string>", line 3, col 8', TypedNode.yaml_loc(inp['foo']['x'], suffix=''))
        self.assertEqual('File "<unicode string>", line 4, col 8: ', TypedNode.yaml_loc(inp['foo']['y']))
        self.assertEqual('File "<unicode string>", line 4, col 8I yam that I yam',
                         TypedNode.yaml_loc(inp['foo']['y'], suffix=inp['foo']['y']))
        self.assertEqual('File "<unicode string>", line 5, col 8: ', TypedNode.yaml_loc(inp['foo']['z']))

        outs = StringIO()
        with redirect_stderr(outs):
            self.assertEqual('File "<unicode string>", line 3, col 8', TypedNode.loc(inp['foo']['x']))
>       self.assertIn('Call to deprecated method loc. (Use yaml_loc instead)', outs.getvalue())
E       AssertionError: 'Call to deprecated method loc. (Use yaml_loc instead)' not found in ''

tests\test_issues\test_issue_6.py:31: AssertionError
________________________________ LoadersUnitTest.test_json_loader ________________________________

self = <tests.test_loaders_dumpers.test_loaders.LoadersUnitTest testMethod=test_json_loader>

    def test_json_loader(self):
        """ Load obo_sample.json, emit obo_sample_json.yaml and check the results """
>       self.loader_test('obo_sample.json', Package, json_loader)

tests\test_loaders_dumpers\test_loaders.py:30:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\test_loaders_dumpers\loaderdumpertestcase.py:63: in loader_test
    self.assertEqual('tests/test_loaders_dumpers/input', os.path.relpath(metadata.base_path, rel_path))
E   AssertionError: 'tests/test_loaders_dumpers/input' != 'tests\\test_loaders_dumpers\\input'
E   - tests/test_loaders_dumpers/input
E   ?      ^                    ^
E   + tests\test_loaders_dumpers\input
E   ?      ^                    ^
________________________________ LoadersUnitTest.test_yaml_loader ________________________________

self = <tests.test_loaders_dumpers.test_loaders.LoadersUnitTest testMethod=test_yaml_loader>

    def test_yaml_loader(self):
        """ Load obo_sample.yaml, emit obo_sample_yaml.yaml and compare to obo_sample_output.yaml """
>       self.loader_test('obo_sample.yaml', Package, yaml_loader)

tests\test_loaders_dumpers\test_loaders.py:26:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\test_loaders_dumpers\loaderdumpertestcase.py:63: in loader_test
    self.assertEqual('tests/test_loaders_dumpers/input', os.path.relpath(metadata.base_path, rel_path))
E   AssertionError: 'tests/test_loaders_dumpers/input' != 'tests\\test_loaders_dumpers\\input'
E   - tests/test_loaders_dumpers/input
E   ?      ^                    ^
E   + tests\test_loaders_dumpers\input
E   ?      ^                    ^
------------------------------------ Captured stderr teardown ------------------------------------ Validation error: File "tests\test_loaders_dumpers\output\dump\obo_sample_context.json", line 1
Stack:  File "C:\Users\david\MyProg\linkml-runtime\tests\support\test_environment.py", line 103 in log
                File "C:\Users\david\MyProg\linkml-runtime\tests\support\test_environment.py", line 241 in eval_single_file
                File "C:\Users\david\MyProg\linkml-runtime\tests\test_loaders_dumpers\loaderdumpertestcase.py", line 33 in dump_test
                File "C:\Users\david\MyProg\linkml-runtime\tests\test_loaders_dumpers\test_dumpers.py", line 55 in test_json_dumper
Output was changed.


Validation error: File "tests\test_loaders_dumpers\output\dumps\obo_sample_context.json", line 1
Stack:  File "C:\Users\david\MyProg\linkml-runtime\tests\support\test_environment.py", line 103 in log
                File "C:\Users\david\MyProg\linkml-runtime\tests\support\test_environment.py", line 241 in eval_single_file
                File "C:\Users\david\MyProg\linkml-runtime\tests\test_loaders_dumpers\loaderdumpertestcase.py", line 45 in dumps_test
                File "C:\Users\david\MyProg\linkml-runtime\tests\test_loaders_dumpers\test_dumpers.py", line 58 in test_json_dumper
Output was changed.
```
</details>

**Warnings**
<details>

```
======================================== warnings summary ======================================== tests\support\test_environment.py:35
  C:\Users\david\MyProg\linkml-runtime\tests\support\test_environment.py:35: PytestCollectionWarning: cannot collect test class 'TestEnvironment' because it has a __init__ constructor (from: tests/support/test_environment.py)
    class TestEnvironment:

.venv\lib\site-packages\_pytest\fixtures.py:229
  C:\Users\david\MyProg\linkml-runtime\.venv\lib\site-packages\_pytest\fixtures.py:229: UserWarning: Code: _pytestfixturefunction is not defined in namespace XSD
    fixturemarker: Optional[FixtureFunctionMarker] = getattr(

tests/test_issues/test_issue_6.py::Issue6TestCase::test_loc_function
  C:\Users\david\MyProg\linkml-runtime\tests\test_issues\test_issue_6.py:30: DeprecationWarning: Call to deprecated method loc. (Use yaml_loc instead)
    self.assertEqual('File "<unicode string>", line 3, col 8', TypedNode.loc(inp['foo']['x']))

tests/test_utils/test_metamodelcore.py::MetamodelCoreTest::test_datetime
  C:\Users\david\MyProg\linkml-runtime\tests\test_utils\test_metamodelcore.py:167: UserWarning: Code: datetime is not defined in namespace XSD
    self.assertEqual('2019-07-06T17:22:39.007300', XSDDateTime(Literal(v, datatype=XSD.datetime)))

tests/test_utils/test_schemaview.py::SchemaViewTestCase::test_schemaview
tests/test_utils/test_schemaview.py::SchemaViewTestCase::test_schemaview
tests/test_utils/test_schemaview.py::SchemaViewTestCase::test_schemaview
  C:\Users\david\MyProg\linkml-runtime\linkml_runtime\utils\schemaview.py:710: DeprecationWarning: Call to deprecated method all_element. (Use 'all_elements' instead)
    elements = self.all_element()
```
</details>

--- a/linkml_runtime/utils/compile_python.py
+++ b/linkml_runtime/utils/compile_python.py
@@ -42,7 +42,7 @@ def compile_python(text_or_fn: str, package_path: str = None) -> ModuleType:
         else:
             warning(f"There is no established path to {package_path} - compile_python may or may not work")
             path_from_tests_parent = os.path.relpath(package_path, os.path.join(os.getcwd(), '..'))
-        module.__package__ = os.path.dirname(os.path.relpath(path_from_tests_parent, os.getcwd())).replace('/', '.')
+        module.__package__ = os.path.dirname(os.path.relpath(path_from_tests_parent, os.getcwd())).replace(os.path.sep, '.')
     sys.modules[module.__name__] = module
     exec(spec, module.__dict__)
     return module

--- a/linkml_runtime/utils/schemaview.py
+++ b/linkml_runtime/utils/schemaview.py
@@ -707,7 +707,7 @@ class SchemaView(object):
 
         """
         applicable_elements = []
-        elements = self.all_element()
+        elements = self.all_elements()
         for category, category_element in elements.items():
             if hasattr(category_element, 'id_prefixes') and prefix in category_element.id_prefixes:
                 applicable_elements.append(category_element.name)

--- a/tests/support/test_environment.py
+++ b/tests/support/test_environment.py
@@ -217,7 +217,7 @@ class TestEnvironment:
         if not self.eval_single_file(expected_file, actual, filtr, comparator):
             if self.fail_on_error:
                 self.make_temp_dir(os.path.dirname(actual_file), clear=False)
-                with open(actual_file, 'w') as actualf:
+                with open(actual_file, 'w', encoding='UTF-8') as actualf:
                     actualf.write(actual)
         return actual
 
@@ -241,7 +241,7 @@ class TestEnvironment:
             self.log(expected_file_path, msg)
         if msg and not self.fail_on_error:
             self.make_temp_dir(os.path.dirname(expected_file_path), clear=False)
-            with open(expected_file_path, 'w') as outf:
+            with open(expected_file_path, 'w', encoding='UTF-8') as outf:
                 outf.write(actual_text)
         return not msg
 

--- a/tests/test_issues/input/issue_368.py
+++ b/tests/test_issues/input/issue_368.py
@@ -20,7 +20,7 @@ from linkml_runtime.utils.formatutils import camelcase, underscore, sfx
 from linkml_runtime.utils.enumerations import EnumDefinitionImpl
 from rdflib import Namespace, URIRef
 from linkml_runtime.utils.curienamespace import CurieNamespace
-from . issue_368_imports import ParentClass, SampleEnum
+from .issue_368_imports import ParentClass, SampleEnum
 
 metamodel_version = "1.7.0"
 

--- a/tests/test_issues/test_issue_6.py
+++ b/tests/test_issues/test_issue_6.py
@@ -1,6 +1,4 @@
 import unittest
-from contextlib import redirect_stderr
-from io import StringIO
 
 import hbreader
 import yaml

--- a/tests/test_issues/test_issue_6.py
+++ b/tests/test_issues/test_issue_6.py
@@ -25,10 +25,10 @@ class Issue6TestCase(unittest.TestCase):
                          TypedNode.yaml_loc(inp['foo']['y'], suffix=inp['foo']['y']))
         self.assertEqual('File "<unicode string>", line 5, col 8: ', TypedNode.yaml_loc(inp['foo']['z']))
 
-        outs = StringIO()
-        with redirect_stderr(outs):
+        with self.assertWarns(DeprecationWarning) as cm:
             self.assertEqual('File "<unicode string>", line 3, col 8', TypedNode.loc(inp['foo']['x']))
-        self.assertIn('Call to deprecated method loc. (Use yaml_loc instead)', outs.getvalue())
+        self.assertEqual('Call to deprecated method loc. (Use yaml_loc instead)', cm.warning.args[0])
+
 
         self.assertEqual('', TypedNode.yaml_loc(None))
         self.assertEqual('', TypedNode.yaml_loc("abc"))

--- a/tests/test_loaders_dumpers/__init__.py
+++ b/tests/test_loaders_dumpers/__init__.py
@@ -1,6 +1,7 @@
 import os
 
 from rdflib import Namespace
+from pathlib import Path
 
 HTTP_TEST_PORT = 8000
 HTTPS_TEST_PORT = 8443
@@ -14,9 +15,9 @@ LD_11_DIR = os.path.join(TESTING_DIR, 'jsonld_context/jsonld_11/')
 
 
 GITHUB_DIR = 'https://raw.githubusercontent.com/HOT-Ecosystem/TermCI-model/main/'
-GITHUB_INPUT_DIR = GITHUB_DIR + os.path.relpath(INPUT_DIR, os.path.dirname(TESTING_DIR))
-GITHUB_LD10_CONTEXT = GITHUB_DIR + os.path.relpath(LD_10_DIR, os.path.dirname(TESTING_DIR)) + '/'
-GITHUB_LD11_CONTEXT = GITHUB_DIR + os.path.relpath(LD_11_DIR, os.path.dirname(TESTING_DIR)) + '/'
+GITHUB_INPUT_DIR = GITHUB_DIR + Path(os.path.relpath(INPUT_DIR, os.path.dirname(TESTING_DIR))).as_posix()
+GITHUB_LD10_CONTEXT = GITHUB_DIR + Path(os.path.relpath(LD_10_DIR, os.path.dirname(TESTING_DIR))).as_posix() + '/'
+GITHUB_LD11_CONTEXT = GITHUB_DIR + Path(os.path.relpath(LD_11_DIR, os.path.dirname(TESTING_DIR))).as_posix() + '/'
 
 SCT = Namespace("http://snomed.info/id/")
 OBO = Namespace("http://purl.obolibrary.org/obo/")

--- a/tests/test_loaders_dumpers/loaderdumpertestcase.py
+++ b/tests/test_loaders_dumpers/loaderdumpertestcase.py
@@ -60,9 +60,14 @@ class LoaderDumperTestCase(TestEnvironmentTestCase):
 
         # Make sure metadata gets filled out properly
         rel_path = os.path.abspath(os.path.join(test_base.env.cwd, '..'))
-        self.assertEqual('tests/test_loaders_dumpers/input', os.path.relpath(metadata.base_path, rel_path))
-        self.assertEqual(f'tests/test_loaders_dumpers/input/{filename}', os.path.relpath(metadata.source_file,
-                                                                                         rel_path))
+        self.assertEqual(
+            os.path.normpath('tests/test_loaders_dumpers/input'), 
+            os.path.normpath(os.path.relpath(metadata.base_path, rel_path))
+        )
+        self.assertEqual(
+            os.path.normpath(f'tests/test_loaders_dumpers/input/{filename}'), 
+            os.path.normpath(os.path.relpath(metadata.source_file, rel_path))
+        )
 
         fileinfo = FileInfo()
         hbread(filename, fileinfo, self.env.indir)

--- a/tests/test_utils/test_metamodelcore.py
+++ b/tests/test_utils/test_metamodelcore.py
@@ -164,7 +164,7 @@ class MetamodelCoreTest(unittest.TestCase):
     def test_datetime(self):
         v = datetime.datetime(2019, 7, 6, 17, 22, 39, 7300)
         self.assertEqual('2019-07-06T17:22:39.007300', XSDDateTime(v))
-        self.assertEqual('2019-07-06T17:22:39.007300', XSDDateTime(Literal(v, datatype=XSD.datetime)))
+        self.assertEqual('2019-07-06T17:22:39.007300', XSDDateTime(Literal(v, datatype=XSD.dateTime)))
         self.assertEqual('2019-07-06T17:22:39.007300', XSDDateTime(Literal(v).value))
         self.assertEqual('2019-07-06T17:22:39.007300', v.isoformat())
         self.assertEqual('2019-07-06T17:22:39.007300', XSDDateTime(XSDDateTime(v)))

